### PR TITLE
Feature/1816 empty state deposits screen

### LINKED ISSUE #1816: Show empty state Deposits screen

### PR DESCRIPTION
**Pull request by moon0326**
Fixes #1816 #1917 

This PR adds empty state Deposits and Transaction screens for the `wcpay_empty_state_preview_mode` experiment.

#### Changes proposed in this Pull Request

1. Created a new directory `empty-state-table` and added any empty state related files and components to minimize changes to the existing codes.
2. Added both Deposits and Transaction screens in this PR since they both share the exact same design, except for the list image.

#### Testing instructions

1. Turn on your A8C proxy.
2. Find `wcpay_empty_state_preview_mode` experiment in Abcus. Add both `control` and `treatment` variations to your bookmark.
3. Open `includes/admin/class-wc-payments-admin.php` file. Find this [line](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/admin/class-wc-payments-admin.php#L88) and add `$should_render_full_menu=true`. This forces WC Payment menu to show up. This change is for testing purposes only and temporary until we have https://github.com/Automattic/woocommerce-payments/issues/1815 ready
4. Login to your WC Admin and assign yourself to `treatment` group by clicking the bookmark.

**Testing Deposits**

1. Navigate to Payments -> Deposits. You should see the preview mode (screenshot #1)
2. Turn on tracking and click the overview area. You should see `wcadmin_wcpay_deposits_summary_empty_state_click` event.
3. Click "Finish setup". You should see `wcadmin_wcpay_connect_account_clicked` event and it should redirects you to WC Payments KYC.

**Testing Transactions**

1. Navigate to Payments -> Deposits. You should see the preview mode (screenshot #2)
2. Click "Finish setup". You should see `wcadmin_wcpay_connect_account_clicked` event and it should redirects you to WC Payments KYC.

**Testing control**

1. Assign yourself to `control` group
2. Navigate to Payments -> Deposits and confirm that you're seeing the default experience.
3. Navigate to Payments -> Transaction and confirm that you're seeing the default experience.

#### Screenshots

Screen Shot 2021-06-12 at 3 04 58 PM
Screen Shot 2021-06-12 at 3 05 03 PM

### CODE REVIEW

**Comment by mattsherman** [comment]
> * Added [empty-state-table](https://github.com/Automattic/woocommerce-payments/blob/feature/1816-empty-state-deposits-screen/client/emtpy-state-table/index.js) component to render an empty table with a component in it. I tried to find a way to inject a component into `TableCard` component in the [list component](https://github.com/Automattic/woocommerce-payments/blob/feature/1816-empty-state-deposits-screen/client/deposits/list/index.js), but it was a bit tricky. Since we're not going to display any data for this time, I figured rendering a table in a separate component is cleaner and easier. This way, we're not editing any of the existing code at least for this time.

Ultimately, I think it would be great to have the `TableCard` component support an "empty state" directly, but I agree that this approach is probably ideal for this initial experiment. It runs the risk of the "empty state" table columns getting out of sync with the real table columns, but I think that is an acceptable risk, as it wouldn't really make much of a difference in practicality.

**Comment by moon0326** [comment]
> > * Added [empty-state-table](https://github.com/Automattic/woocommerce-payments/blob/feature/1816-empty-state-deposits-screen/client/emtpy-state-table/index.js) component to render an empty table with a component in it. I tried to find a way to inject a component into `TableCard` component in the [list component](https://github.com/Automattic/woocommerce-payments/blob/feature/1816-empty-state-deposits-screen/client/deposits/list/index.js), but it was a bit tricky. Since we're not going to display any data for this time, I figured rendering a table in a separate component is cleaner and easier. This way, we're not editing any of the existing code at least for this time.
> 
> Ultimately, I think it would be great to have the `TableCard` component support an "empty state" directly, but I agree that this approach is probably ideal for this initial experiment. It runs the risk of the "empty state" table columns getting out of sync with the real table columns, but I think that is an acceptable risk, as it wouldn't really make much of a difference in practicality.

Thank you @mattsherman for the feedback. I have one more question. I left the same question on [here](https://github.com/Automattic/woocommerce-payments/issues/1816). Is the ellipsis button functional? I was going to use this simple HTML version of `TableCard`, but we need to modify `TableCard` if we're going to support ellipsis button.

File: client/emtpy-state-table/list.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+			wpcom_connection: wcpaySettings.isJetpackConnected ? 'Yes' : 'No',
+		} );
+	};
+
+	return (
+		<div className="empty-state-deposits-list">
+			<div>
+				<img src={ props.listBanner } alt="" />
+			</div>
+			<p className="intro-copy">
```
Comment (adrianduffell) [inline client/emtpy-state-table/list.js:45]: This text isn't i18n.

**Review by adrianduffell — CHANGES_REQUESTED** [review]
@moon0326 This is looking good, nice job on this 👍 

I just noticed a few places where the text isn't using the `__()` i18n function.

**Comment by elizaan36** [comment]
hey @moon0326 was passing by to take a look. Looks great, just have a few notes.

- In the design, the illustration doesn't have the black outlines I'm seeing in the screenshots you shared above. I'll share links to the SVG for each illustration here in case this helps. [Deposits](https://share.getcloudapp.com/rRujobmd) | [Transactions](https://share.getcloudapp.com/bLukwqyP)
Screen Shot 2021-06-14 at 3 41 04 PM

- Is it possible for the Date column background color to match the color of the rest of the table header (also shown in the example above)
- The container could use 20px padding on top and bottom (above illustrations and below CTA)
- Lastly, this is more of a question for @samueljseay on the Overview screen. I had a look around but couldn't find anything - is there an issue for presenting the deposits summary on the Payments overview screen at all?

Once the Payments overview screen is implemented based on the designs in paJDYF-18j-p2, the summary number components at the top of the deposit screens won't be necessary since that information will be on the overview. If that will be rolled out relatively soon we might be able to remove the summary numbers from the Deposits screen and this empty state.

**Comment by moon0326** [comment]
> hey @moon0326 was passing by to take a look. Looks great, just have a few notes.
> 
> * In the design, the illustration doesn't have the black outlines I'm seeing in the screenshots you shared above. I'll share links to the SVG for each illustration here in case this helps. [Deposits](https://share.getcloudapp.com/rRujobmd) | [Transactions](https://share.getcloudapp.com/bLukwqyP)
> 
> Screen Shot 2021-06-14 at 3 41 04 PM
> 
> * Is it possible for the Date column background color to match the color of the rest of the table header (also shown in the example above)
> * The container could use 20px padding on top and bottom (above illustrations and below CTA)
> * Lastly, this is more of a question for @samueljseay on the Overview screen. I had a look around but couldn't find anything - is there an issue for presenting the deposits summary on the Payments overview screen at all?
> 
> Once the Payments overview screen is implemented based on the designs in paJDYF-18j-p2, the summary number components at the top of the deposit screens won't be necessary since that information will be on the overview. If that will be rolled out relatively soon we might be able to remove the summary numbers from the Deposits screen and this empty state.

Thank you @elizaan36 for the review 👍 I will make the changes.

File: client/emtpy-state-table/list.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+				Collect payments, track cash flow, and manage recurring revenue
+				directly from your store&apos;s dashboard -- all without setup
+				costs or monthly fees.
+			</p>
+			<p className="terms-of-service">
+				By clicking &quot;Finish setup,&quot; you agree to the{ ' ' }
```
[the remark this answers is not in the document]
Comment (moon0326) [reply client/emtpy-state-table/list.js]: Updated in https://github.com/Automattic/woocommerce-payments/pull/1963/commits/ab1e7a344ed5a07a09ef5939c7ebbe3df1b86043

**Comment by moon0326** [comment]
@elizaan36 

Updated your changes.

New screenshots:

Screen Shot 2021-06-14 at 5 31 21 PM
Screen Shot 2021-06-14 at 5 31 17 PM

**Comment by elizaan36** [comment]
Perfect, thanks Moon!
And I just realized that the Overview screen was released in the last version with the Deposits summary so we're clear to hide the summary number row at the top of the Deposits screen.

File: client/emtpy-state-table/list.js (hunk 1 above)
Comment (adrianduffell) [reply client/emtpy-state-table/list.js:46]: Thanks @moon032! Looks like the `'woocommerce-payments'` text domain was missed on the intro copy text.

**Comment by moon0326** [comment]
> Perfect, thanks Moon!
> And I just realized that the Overview screen was released in the last version with the Deposits summary so we're clear to hide the summary number row at the top of the Deposits screen.

Even better! Removed the overview section.

Screen Shot 2021-06-14 at 6 36 56 PM

File: client/emtpy-state-table/list.js (hunk 1 above)
Comment (moon0326) [reply client/emtpy-state-table/list.js:46]: @adrianduffell Thank you for catching it 👍  Updated in  https://github.com/Automattic/woocommerce-payments/pull/1963/commits/a52ce4a107ef9c24cd4a4ce50319112e45f80cf6

I also removed the overview page on the deposits screen as per Elizabeth in https://github.com/Automattic/woocommerce-payments/pull/1963/commits/d2d767bf1934f2f199ac8114f9823378793940bc

**Comment by adrianduffell** [comment]
> This change is for testing purposes only and temporary until we have #1815 ready

@moon0326 I've got a PR for this now https://github.com/Automattic/woocommerce-payments/pull/2253

**Comment by adrianduffell** [comment]
@moon0326 The PHP ExPlat client in PR #2253 is now merged. You should be able to integrate it with something like:

```php
$abtest = new \WCPay\Experimental_Abtest(
	$_COOKIE[ 'tk_ai' ],
	'woocommerce',
	'yes' === get_option( 'woocommerce_allow_tracking' )
);

$should_render_full_menu = ( 'treatment' === $abtest->get_variation( 'wcpay_empty_state_preview_mode' )  );
```

**Comment by moon0326** [comment]
> @moon0326 The PHP ExPlat client in PR #2253 is now merged. You should be able to integrate it with something like:
> 
> ```
> $abtest = new \WCPay\Experimental_Abtest(
> 	$_COOKIE[ 'tk_ai' ],
> 	'woocommerce',
> 	'yes' === get_option( 'woocommerce_allow_tracking' )
> );
> 
> $should_render_full_menu = ( 'treatment' === $abtest->get_variation( 'wcpay_empty_state_preview_mode' )  );
> ```

Thank you @adrianduffell 👍 I will make the change.

**Comment by moon0326** [comment]
> @moon0326 The PHP ExPlat client in PR #2253 is now merged. You should be able to integrate it with something like:
> 
> ```
> $abtest = new \WCPay\Experimental_Abtest(
> 	$_COOKIE[ 'tk_ai' ],
> 	'woocommerce',
> 	'yes' === get_option( 'woocommerce_allow_tracking' )
> );
> 
> $should_render_full_menu = ( 'treatment' === $abtest->get_variation( 'wcpay_empty_state_preview_mode' )  );
> ```

@adrianduffell Added ExPlat in https://github.com/Automattic/woocommerce-payments/pull/1963/commits/c02ed43a3fb7d419bf43d40f009a3e6f6244d187

I wasn't able to test it on the server-side though. 

I keep getting empty string from `$abtest->get_variation( 'wcpay_empty_state_preview_mode' )` call.

I've tried with proxied API endpoint and my site is connected to Jetpack. 

Is there a condition I need to set before testing the server side?

**Review by adrianduffell — APPROVED** [review]
@moon0326 Nice work on this. Let's 🚢 

Thanks for adding ExPlat. I think the `null` value is an issue in the `Experimental_Abtest` client when the experiment hasn't launched yet. I think a cookie needs to be sent relating to the bookmarklet assignments. I'm investigating further.

**Pull request merged**